### PR TITLE
(newapp) Improve Form typing: remove need to pass schema type

### PR DESCRIPTION
**Pull request by flybayer**
### What are the changes and their implications?

Improve Form typing for new apps: remove need to pass schema type

### CODE REVIEW

File: examples/auth/app/components/Form.tsx [hunk 1]
```diff
@@ -3,26 +3,26 @@ import {Form as FinalForm, FormProps as FinalFormProps} from "react-final-form"
...
+  onSubmit: FinalFormProps<z.infer<S>>["onSubmit"]
+  initialValues?: FinalFormProps<z.infer<S>>["initialValues"]
+  schema?: S
 } & Omit<PropsWithoutRef<JSX.IntrinsicElements["form"]>, "onSubmit">
 
-export function Form<FormValues extends Record<string, unknown>>({
+export function Form<S extends z.ZodType<any, any>>({
   children,
   submitText,
   schema,
```
Comment (nitaking) [inline examples/auth/app/components/Form.tsx:19]: Which one is more type-checked?

`ZodType<any, any>` `Record<string, unknown>`


It seems that unknown is better than any

Comment (flybayer) [reply examples/auth/app/components/Form.tsx:19]: Thanks for reviewing!

In this case `S` is **extending** `z.ZodType<any, any>` which means that it's up to the user to define what the actual type will be. So it doesn't matter what we use here.

Comment (nitaking) [reply examples/auth/app/components/Form.tsx:19]: I see! I'll try it next time. Thanks for the comment!

**Comment by nitaking** [comment]
LGTM!

**Pull request merged**